lib/chef/resource/chef_client_scheduled_task.rb

```diff
@@ -136,7 +138,7 @@ class ChefClientScheduledTask < Chef::Resource
 
       property :chef_binary_path, String,
         description: "The path to the #{ChefUtils::Dist::Infra::CLIENT} binary.",
-        default: "C:/#{ChefUtils::Dist::Org::LEGACY_CONF_DIR}/#{ChefUtils::Dist::Infra::DIR_SUFFIX}/bin/#{ChefUtils::Dist::Infra::CLIENT}"
+        default: lazy { Chef::ResourceHelpers::PathHelpers.chef_client_hab_binary_path }
```

Comment:
This doesn't seem like what we want. This means that if I install a newer chef hab package to test, it'll suddenly move everything to it.

This should point to a common bin path, which the user can then override when ready.

Reply:
But then if I'm on version X, and I install version Y to do some testing, you automatically move me.

Wouldn't it make more sense to determine the version of hab infra client we're running under and use _that_ version?

Reply:
To be more specific - I have version 19.8 installed sometime int he future.

Now install version 20rc1 next to it just to test some stuff, and bam my next chef 19 run MOVES MY SYSTEM to Chef 20! Seems very wrong.

Reply:
Or worse yet, it actually goes by mtime, not version string, so if I install some older version to check something in whyrun mode, and the next chef run moves my chef crons to that old version. That's definitely bad.

Reply:
To expand upon that though the same code in 20.x would then replace the binary path when running the test. I think we need an option to specify the version number and if not specified default to the current running chef-client version. Users can then assign it based on the attribute for specified infra-client they expect to be installed if using a CB like chef_client_updater CB.